doc/source/io.rst

```diff
@@ -1175,6 +1175,80 @@ data that appear in some lines but not others:
     0  1  2   3
     1  4  5   6
     2  8  9  10
+    
+Handling "bad" lines - preserving the data
+''''''''''''''''''''
+    
+To preserve all data, you can specify header ``names`` that are long enough:
+
+.. code-block:: ipython
+
+   In [31]: pd.read_csv(StringIO(data), names=['a', 'b', 'c', 'd'])
+   
+   Out[31]:
+       a  b   c  d
+    0  1  2   3  NaN
+    1  4  5   6  7
+    2  8  9  10  NaN
+    
+or you can use Python's ``open`` command to detect the length of the widest row:
```

Comment:
I'm not sure you need to include this workaround.  @jorisvandenbossche suggestion, which you added above, should suffice.

Reply:
Note, that you can always over-specify the length and cut down later, which is a lot less suffering even than this, even though I proposed it :smile:

Reply:
I think the generation of arbitrary labels is generally an approach that should be avoided:

- To others reviewing the code, the DataFrame, or a potentially exported file, it implies some level of purpose, or customization in label specification. They might seek to understand why these labels were used, and if there's a matching codebook or other DataFrames.
- It's messy. Using a `for` loop to create strings for the number of variables necessary doesn't feel like something that should be required.

So I like including at least the first ``open`` example, because it's the only way to preserve all data without customizing header labels.

Reply:
Not really in fact.  Just use `names = ['dummy'] * n`, and we'll take of the duplicates for you.

Reply:
@gfyoung Can you clarify? Where's the value for `n` coming from?

Reply:
@joshjacobson : Replace `n` with whatever number of columns you want.  This is a placeholder.